stabby-macros: rand::random in import macro makes all dependents non-reproducible

## Problem

`stabby-macros/src/functions.rs` uses `rand::random::<u128>()` to generate
internal module identifiers for `#[stabby::import]` extern blocks:

```rust
let modid = quote::format_ident!("_stabbymod_{}", rand::random::<u128>());
```

This makes every crate compiled with stabby macros (e.g. `zenoh-plugin-trait`,
`zenoh`) non-reproducible. Each build produces a different rlib with a different
baked-in module hash, which poisons binary caches: any build that mixes a
cache-substituted zenoh artifact with a locally-rebuilt downstream crate hits:

```
error[E0460]: found possibly newer version of crate `stabby` which `zenoh` depends on
error[E0463]: can't find crate for `zenoh`
```

---

**Note: this only manifests in deterministic/cached build environments**

In a standard `cargo build` workflow, all crates in the workspace compile together in a single consistent pass. Every `rand::random::<u128>()` call produces a fresh value, but all crates in that build see mutually consistent values — so nothing breaks.

The problem surfaces only in **binary cache environments like Nix**, where:

1. Individual crate build outputs are cached and keyed by their derivation hash (a pure function of inputs).
2. A cached `zenoh` artifact (built on CI in one consistent pass) is stored with its `_stabbymod_<X>` hash baked in.
3. A downstream crate (`crate-x`) is later **rebuilt locally** — e.g. because an unrelated upstream dep changed — and links against the **cached** `zenoh`.
4. The locally-rebuilt `crate-x` gets a fresh `_stabbymod_<Y>` (Y ≠ X), so rustc rejects the cached `zenoh` rlib with `E0460`/`E0463`.

The result is a cache poisoning cycle: CI fails because it also rebuilds incrementally against the cached zenoh, the branch outputs never make it into the cache, and every developer who checks out the branch hits the same failure.

This would apply to any binary cache system that stores compiled crate artifacts and performs incremental builds against them (Nix, Bazel remote cache, sccache in certain modes, etc.). The fix makes the macro output stable so cached artifacts remain valid across incremental rebuilds.
## Fix

Replace \`rand::random\` with a deterministic polynomial hash of the full token
stream of the extern block's items. Unique per call site, removes the \`rand\`
dependency from \`stabby-macros\` entirely.

PR: https://github.com/ZettaScaleLabs/stabby/pull/136

--- a/stabby-macros/Cargo.toml
+++ b/stabby-macros/Cargo.toml
@@ -34,7 +34,6 @@ proc-macro2 = { workspace = true }
 proc-macro-crate = { workspace = true }
 quote = { workspace = true }
 syn = { workspace = true, features = ["full", "extra-traits"] }
-rand = { workspace = true }
 
 [lib]
 proc-macro = true

--- a/stabby-macros/src/functions.rs
+++ b/stabby-macros/src/functions.rs
@@ -429,7 +429,7 @@ pub fn import(
         attrs, abi, items, ..
     } = &fn_decl;
     let st = crate::tl_mod();
-    let modid = quote::format_ident!("_stabbymod_{}", rand::random::<u128>());
+    let modid = quote::format_ident!("_stabbymod_{}", quote::quote!(#(#items)*).to_string().bytes().fold(0u128, |a, b| a.wrapping_mul(31).wrapping_add(b as u128)));
     let mut externs = Vec::new();
     let mut interns = Vec::new();
     let mut intern_ids = Vec::new();

--- a/stabby-macros/src/functions.rs
+++ b/stabby-macros/src/functions.rs
@@ -429,7 +429,13 @@ pub fn import(
         attrs, abi, items, ..
     } = &fn_decl;
     let st = crate::tl_mod();
-    let modid = quote::format_ident!("_stabbymod_{}", quote::quote!(#(#items)*).to_string().bytes().fold(0u128, |a, b| a.wrapping_mul(31).wrapping_add(b as u128)));
+    let modid = {
+        use std::collections::hash_map::DefaultHasher;
+        use std::hash::{Hash, Hasher};
+        let mut hasher = DefaultHasher::new();
+        quote::quote!(#(#items)*).to_string().hash(&mut hasher);
+        quote::format_ident!("_stabbymod_{:x}", hasher.finish())
+    };
     let mut externs = Vec::new();
     let mut interns = Vec::new();
     let mut intern_ids = Vec::new();

